### PR TITLE
20231025 attestation ca devices

### PR DESCRIPTION
**Pull request by Firstyear**
This finishes some of the last details to make attestation usable in external applications. It improves the AttestationCaList type to include localised strings of accepted devices for UI prompts. It creates an AttestationCaListbuilder to allow custom lists to be crafted. It also cleans up some of the elements of the fido mds tool output to aid users in querying and creating these lists. It also changes the demo site to use the fido mds as an attestation option. 

- [ x ] cargo test has been run and passes
- [ x ] documentation has been updated with relevant examples (if relevant)

### CODE REVIEW

File: fido-key-manager/src/main.rs [hunk 1]
```diff
@@ -13,7 +13,6 @@ use std::time::Duration;
 use tokio_stream::StreamExt;
 #[cfg(feature = "solokey")]
 use webauthn_authenticator_rs::ctap2::SoloKeyAuthenticator;
-use webauthn_authenticator_rs::prelude::WebauthnCError;
```
Comment (micolous) [inline fido-key-manager/src/main.rs:16]: Gate on `#[cfg(feature = "solokey")]` to resolve the warning and not break the build

File: attestation-ca/src/lib.rs [hunk 2]
```diff
@@ -99,79 +114,17 @@ pub struct AttestationCaList {
     pub cas: BTreeMap<Base64UrlSafeData, AttestationCa>,
 }
 
-impl TryFrom<AttestationCa> for AttestationCaList {
```
Comment (micolous) [inline attestation-ca/src/lib.rs:102]: Update `webauthn-authenticator-rs/src/softtoken.rs:932:10` and `webauthn-authenticator-rs/src/softtoken.rs:1014:10` for this

File: attestation-ca/src/lib.rs [hunk 3]
```diff
@@ -24,11 +40,11 @@ pub struct SerialisableAttestationCa {
 )]
 pub struct AttestationCa {
     /// The x509 root CA of the attestation chain that a security key will be attested to.
-    pub ca: x509::X509,
+    ca: x509::X509,
     /// If not empty, the set of acceptable AAGUIDS (Device Ids) that are allowed to be
     /// attested as trusted by this CA. AAGUIDS that are not in this set, but signed by
     /// this CA will NOT be trusted.
-    pub aaguids: BTreeSet<Uuid>,
+    aaguids: BTreeMap<Uuid, DeviceDescription>,
```
Comment (micolous) [inline attestation-ca/src/lib.rs:47]: Update `webauthn-authenticator-rs/src/softtoken.rs:930:13` and `webauthn-authenticator-rs/src/softtoken.rs:1012:13` for this

**Review by micolous — CHANGES_REQUESTED** [review]
Many CI detected failures

**Comment by micolous** [comment]
I think wait for #392; that'll make CI green again :)

**Comment by Firstyear** [comment]
yep, no stress.

File: fido-mds/src/lib.rs [hunk 4]
```diff
@@ -1424,19 +1447,26 @@ impl FidoMds {
...
+
+                att_ca_builder
+                    .insert_device_der(
+                        ca.as_slice(),
+                        fd.aaguid,
+                        fd.description.clone(),
+                        fd.alternative_descriptions.clone(),
+                    )
+                    .map_err(|e| {
+                        error!(err = ?e, "Failed to add FIDO2 device to attestation ca list");
```
Comment (micolous) [inline fido-mds/src/lib.rs:1464]: `.map_err(|err| { ...` then `error!(?err, "Failed ...`

File: fido-mds/src/mds.rs [hunk 5]
```diff
@@ -488,6 +499,18 @@ pub enum AuthenticatorTransport {
     Internal,
 }
 
+impl fmt::Display for AuthenticatorTransport {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            AuthenticatorTransport::Usb => write!(f, "usb"),
```
Comment (micolous) [inline fido-mds/src/mds.rs:505]: Are these intended to be human-readable values, like `AuthenticatorVersion`?

If so, they should be capitalized appropriately, and `FromStr for AuthenticatorTransport` should do case-insensitive matching.

File: attestation-ca/src/lib.rs [hunk 6]
```diff
@@ -2,15 +2,31 @@ use base64urlsafedata::Base64UrlSafeData;
...
+    pub(crate) en: String,
+    pub(crate) localised: BTreeMap<String, String>,
+}
+
+impl DeviceDescription {
+    pub fn description_en(&self) -> &str {
+        self.en.as_str()
+    }
+
+    pub fn description_localised(&self) -> &BTreeMap<String, String> {
```
Comment (micolous) [inline attestation-ca/src/lib.rs:20]: Describe the keys of this map here. I think they're derived verbatim from the MDS (`alternative_descriptions`), and they're never used in test data.

**Review by micolous — APPROVED** [review]

File: fido-mds/src/mds.rs (hunk 5 above)
Comment (Firstyear) [reply fido-mds/src/mds.rs:505]: I think having them match the "standard" definition of the transport is better.

**Review by micolous — APPROVED** [review]

**Pull request merged**